src/app.js

```diff
@@ -1,20 +1,89 @@
-const express = require('express');
-const app = express();
+require('dotenv').config()
+require('express-async-errors')
+const express = require('express')
+const session = require('express-session')
+const app = express()
+const morgan = require('morgan')
+const bodyParser = require('body-parser')
 const cors = require('cors')
-const favicon = require('express-favicon');
-const logger = require('morgan');
+const favicon = require('express-favicon')
+const { xss } = require('express-xss-sanitizer')
+const helmet = require('helmet')
+const passport = require('passport')
+const mongoose = require('mongoose')
+mongoose.set('strictQuery', true)
+const cookieParser = require('cookie-parser')
+const rateLimit = require('express-rate-limit')
+const limiter = rateLimit({
+    windowMs: 15 * 60 * 1000, // 15 minutes
+    max: 60, // limit each IP to 100 requests per windowMs
+})
 
-const mainRouter = require('./routes/mainRouter.js');
+const logger = require('./logger')
 
-// middleware
-app.use(cors());
-app.use(express.json());
-app.use(express.urlencoded({ extended: false }));
-app.use(logger('dev'));
+app.use(limiter)
+// middleware setup
+
+app.use(xss())
+
+app.use(express.json())
+//Security middleware
+app.use(helmet())
+
+app.use(cors({ origin: [/localhost:8000$/], credentials: true }))
+
+//Logging middleware (using morgan)
+app.use(
+    morgan('dev', {
+        stream: { write: (message) => logger.info(message.trim()) },
+    })
+)
 app.use(express.static('public'))
-app.use(favicon(__dirname + '/public/favicon.ico'));
+app.use(favicon(__dirname + '/public/favicon.ico'))
+app.use(bodyParser.json())
+
+app.use(cookieParser())
+// Configure express-session middleware
+app.use(
+    session({
+        secret: 'your-secret-key', // Replace with a secret key for session data encryption
+        resave: false,
+        saveUninitialized: false,
+        // Other configuration options can be added as needed
+    })
+)
+
+app.use(passport.initialize())
+app.use(passport.session())
+
+// Database setup (using Mongoose)
+mongoose.set('strictQuery', true)
+
+const connectDB = (url) => {
+    return mongoose.connect(url)
+}
+
+module.exports = connectDB
+
+//routers
+const authRouter = require('./routes/authRoutes.js')
+
+app.use('/api/v1/auth', authRouter)
+
+// Error handling middleware (must be defined after all other routes and middleware)
+//add later
 
-// routes
-app.use('/api/v1', mainRouter);
+// Start the server
+const port = process.env.PORT || 3000
```

Comment:
I'm sorry I think I miscommunicated here. We can keep the **backend** server's port as `8000`, but the **frontend** repo will run locally on `3000`.

Because of that we need to make our CORS middleware, allow requests _from_ port 3000, so we need to update line 33 to allow that:

```
app.use(cors({ origin: [/localhost:3000$/], credentials: true }))
```

and revert this line back to how it was before:

```suggestion
const port = process.env.PORT || 8000
```

Reply:
done